doc/source/install/cmake.rst

```diff
@@ -0,0 +1,253 @@
+for more details.
+
+Here is an example how to enable the mruby support.
+
+.. code-block:: console
+
+   $ cmake . -DGRN_WITH_MRUBY=ON
+
+``-DGRN_WITH_DEBUG``
+^^^^^^^^^^^^^^^^^^^^
+
+Enables debug options for C/C++ compiler. It's useful for debugging on debugger such as GDB and LLDB.
+
+The default is ``OFF``.
+
+Here is an example how to enable debug options.
+
+.. code-block:: console
+
+   $ cmake . -DGRN_WITH_DEBUG=ON
+
+``-DGRN_WITH_APACHE_ARROW``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Enables Apache Arrow support.
+
+In addition to using Apache Arrow IPC streaming format output, you can also use multithreading processing that is used in :ref:`select-n-workers` 
+and :doc:`/reference/functions/query_parallel_or` with the Apache Arrow support.   
+
+The default is ``OFF``.
+
+You can install Apache Arrow following to `the official installation procedure <https://arrow.apache.org/install/>`_. 
+
+Here is an example how to enable the Apache Arrow support.
+
+.. code-block:: console
+
+   $ cmake . -DGRN_WITH_APACHE_ARROW=ON
+
+.. note::
+
+   If you install Apache Arrow manually, you need to use the :ref:`cmake-options-cmake-prefix-path` option.
+
+.. _cmake-options-cmake-prefix-path:
+
+``-DCMAKE_PREFIX_PATH=PATHS``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Adds search paths for ``.cmake`` files.
+
+You can specify multiple path separating them with ``:`` on GNU/Linux or UNIX, ``;`` on Windows.
+
+In case of using libraries installed via a package manager, you do not need to specify this 
+parameter. It is because ``.cmake`` files for those libraries are in the default search paths of CMake.
+
+In case of using libraries installed in non-system directories such as ``/usr``, you need to specify ``.cmake`` file paths of those libraries by this parameter.
+
+Here is an example how to specify a ``.cmake`` file path for ``/tmp/local/lib/cmake/Arrow/ArrowConfig.cmake`` on GNU/Linux or UNIX.
+
+.. code-block:: console
+
+   $ cmake . -DCMAKE_PREFIX_PATH="/tmp/local/lib/cmake/Arrow"
+
+Here is an example how to specify a ``.cmake`` file path for ``C:\arrow\lib\cmake\Arrow\ArrowConfig.cmake`` on Windows.
+
+.. code-block:: pwsh-session
+
+   > cmake . -DCMAKE_PREFIX_PATH="C:\arrow\lib\cmake\Arrow"
+
+.. _cmake-build-and-install:
+
+Build and install Groonga
+-------------------------
+
+Now, you can build Groonga.
+
+GNU/Linux or UNIX
++++++++++++++++++
+
+you can use ``make``.
+
+Here is a command line to build and install Groonga by ``make``.
+
+.. code-block:: console
+
+  $ make -j8
+  $ sudo make install
+
+``-j8`` decreases build time. It enables parallel build. If you have 8
+or more CPU cores, you can increase ``8`` to decreases more build
+time.
+
+You can just see only warning and error messages by ``> /dev/null``.
+
+.. code-block:: console
+
+  $ make -j8 > /dev/null
+
+Windows
++++++++
+
+You can use Visual Studio or ``cmake --build``.
+
+Here is a command line to build and install Groonga by ``cmake --build``.
```

Comment:
FYI: We can use `cmake --build` for `Unix Makefiles` too. We may unify build step description in the future.

Reply:
We can use the `CMAKE_BUILD_PARALLEL_LEVEL` environment variable for parallel build with ``cmake --build`: https://cmake.org/cmake/help/latest/envvar/CMAKE_BUILD_PARALLEL_LEVEL.html

Reply:
Ok, I will add description about `CMAKE_BUILD_PARALLEL_LEVEL` and `--parallel [<jobs>], -j [<jobs>]` (https://cmake.org/cmake/help/latest/manual/cmake.1.html#build-tool-mode) in the future.